### PR TITLE
Add provider for custom fields API

### PR DESCRIPTION
**Pull request by mfatyga-sumo**

### CODE REVIEW

File: sumologic/resource_sumologic_field.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceSumologicField() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicFieldCreate,
+		Read:   resourceSumologicFieldRead,
+		Update: resourceSumologicFieldUpdate,
```
Comment (seansain) [inline sumologic/resource_sumologic_field.go:13]: Since update is `nil` we can simply remote it instead. Similar to how it was in content: https://github.com/SumoLogic/terraform-provider-sumologic/blob/5501c3e68ff12e2bdf6a03a7d227e7247d09fbe8/sumologic/resource_sumologic_content.go

**Comment by seansain** [comment]
You will need to add this resource to the `provider.go` file for it to be picked up by the provider.

**Comment by seansain** [comment]
Would be nice to get some docs too.

**Comment by mfatyga-sumo** [comment]
> Would be nice to get some docs too.

What form of documentation do you think would be appropriate, godoc documentation or something else?

File: sumologic/resource_sumologic_field.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+			"field_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: false,
+			},
+
+			"data_type": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (frankreno) [inline sumologic/resource_sumologic_field.go:34]: Why is this required? We don't require this on the create API call.

File: sumologic/resource_sumologic_field.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				ForceNew: true,
+			},
+
+			"state": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (frankreno) [inline sumologic/resource_sumologic_field.go:40]: Same comment as above...

**Review by frankreno — APPROVED** [review]
LGTM, pending docs and questions on the 2 required fields.

**Comment by frankreno** [comment]
> > Would be nice to get some docs too.
> 
> What form of documentation do you think would be appropriate, godoc documentation or something else?

See the website/docs folder in root of readme

**Comment by mfatyga-sumo** [comment]
Please take another look.
